Add AppVeyor CI configuration

cthoyt opened the pull request:
This PR will be useful for addressing #92, which is likely a Windows-only problem. It does the following:

- adds the appropriate AppVeyor configuration for automating testing on Windows
- fixes a unicode loading problem with numpy that only happens on windows
- adds user documentation for installation on Windows with Anaconda

cthoyt committed:
Add appveyor configuration

lvermue committed:
Merge branch 'master' into add-appyveyor

lvermue committed:
Exclude appveyor.yml from MANIFEST

lvermue committed:
Fix Windows specific PyTorch installation issue

lvermue committed:
Fix Windows specific PyTorch installation issue

lvermue committed:
Fix Windows specific PyTorch installation issue

lvermue committed:
Fix Windows specific torch installation error

lvermue committed:
Redesigned AppVeyor configuration file

lvermue committed:
Redesigned AppVeyor configuration file

lvermue committed:
Redesigned AppVeyor configuration file

lvermue committed:
Redesigned AppVeyor configuration file

cthoyt commented on .appveyor.yml:
```diff
@@ -0,0 +1,33 @@
+  - cmd: conda install pytorch torchvision cudatoolkit=10.2 -c pytorch
+
+test_script:
+  - "%CMD_IN_ENV% tox"
```
can specify the environments `tox -e py,integration` so it doesn't run all of the other tests that aren't really relevant here (like documentation build, flake8, etc.)

cthoyt replied:
does `%CMD_IN_ENV%` make sure that the tox is installed on the conda environment?

lvermue replied:
No, I installed it explicitly in the environment under install.
`%CMD_IN_ENV%` just ensures that the command is run within the current environment.

lvermue committed:
Add correct Windows environment handling in AppVeyor

lvermue committed:
Add AppVeyor python version checking

lvermue committed:
Add correct Windows Miniconda Environment to AppVeyor config

lvermue committed:
Testing test_script handling in AppVeyor

lvermue committed:
AppVeyor installation script testing

lvermue committed:
Testing test_scripts with AppVeyor

lvermue committed:
Update AppVeyor config

lvermue committed:
Fix AppVeyor config

lvermue committed:
Fix AppVeyor config

lvermue committed:
Finalize AppVeyor config

lvermue committed:
Finalize AppVeyor config

lvermue committed:
Finalize AppVeyor config

lvermue committed:
Explicitly install SQLite in conda env

lvermue committed:
AppVeyor testing

lvermue committed:
AppVeyor testing

lvermue committed:
Add conda DLLs to Windows PATH

lvermue committed:
Checking for sqlite3 DLL

lvermue committed:
Windows conda PATH checking

cthoyt commented on .appveyor.yml:
```diff
@@ -24,14 +24,15 @@ install:
 
 test_script:
-  - tox -e py,integration
+  - coverage run -p -m pytest --durations=20 tests -m slow
```
we can even skip the `coverage run -p -m` part i the beginning

lvermue committed:
Add conda activation to AppVeyor config

lvermue committed:
Finalize AppVeyor config

lvermue committed:
Finalize AppVeyor config

cthoyt committed:
Update .appveyor.yml

- coverage isn't really needed
- better splitting up of the fast and slow tests
- fix bash-style configuation default `{posargs:test}` that probably breaks on windows
- use pip to install some optional dependencies that are used during testing

cthoyt committed:
Remove installation of coverage

lvermue committed:
Ensure Unicode encoding when loading triples from txt files

cthoyt committed:
Make it possible to specify encodings

cthoyt committed:
Update installation.rst

cthoyt committed:
Add appveyor badge

cthoyt committed:
Update README.md

[skip ci]

lvermue committed:
Cleanup Windows PATH in AppVeyor config

cthoyt committed:
Update README.md

lvermue committed:
Update README.md

cthoyt committed:
Fix links on readme

lvermue committed:
Fix links on readme to organizational setup
